examples/meta/src/model_selection/grid_search.sg

```diff
@@ -0,0 +1,71 @@
+File f_feats_train = csv_file("../../data/classifier_binary_2d_nonlinear_features_train.dat")
+File f_feats_test = csv_file("../../data/classifier_binary_2d_nonlinear_features_test.dat")
+File f_labels_train = csv_file("../../data/classifier_binary_2d_nonlinear_labels_train.dat")
+File f_labels_test = csv_file("../../data/classifier_binary_2d_nonlinear_labels_test.dat")
+
+#![create_features]
+Features features_train = features(f_feats_train)
+Features features_test = features(f_feats_test)
+Labels labels_train = labels(f_labels_train)
+Labels labels_test = labels(f_labels_test)
+#![create_features]
+
+#![create_kernel]
+Kernel k1 = kernel("GaussianKernel")
+Kernel k2 = kernel("PolyKernel")
+#![create_kernel]
+
+#![create_machine]
+Machine svm = machine("LibSVM", epsilon=0.001)
+#![create_machine]
+
+#![evaluation_criteria]
+SplittingStrategy splitting = splitting_strategy("CrossValidationSplitting", num_subsets=4)
+Evaluation evaluation_criterion = evaluation("AccuracyMeasure")
+#![evaluation_criteria]
+
+#![create_master_node]
+GridSearch node = new GridSearch(svm, splitting, evaluation_criterion)
+RealVector c1_param(3)
+c1_param[0] = 0.1
+c1_param[1] = 1
+c1_param[2] = 10
+RealVector c2_param(3)
+c2_param[0] = 0.1
+c2_param[1] = 1
+c2_param[2] = 10
+node.attach("C1", c1_param)
+node.attach("C2", c2_param)
+#![create_master_node]
+
+#![create_child_nodes]
+GridParameters k1_node = new GridParameters(k1)
```

Comment:
I am still a bit confused why the type of the base node and the sub-nodes is different?

Reply:
Hmm, maybe can discuss better this week on irc or something. But basically I don't want each node to have a train method, so I attach nodes to the GridSearch, which has can then be trained.

Reply:
yeah ok let's discuss in irc. But re your comment I would say that the tree would be something that is passed to the grid-search strategy and the latter has the train method and problem solved or?